pandas/core/indexes/base.py

```diff
@@ -5479,9 +5465,10 @@ def _cmp_method(self, other, op):
         """
         Wrapper used to dispatch comparison operations.
         """
-        if isinstance(other, (np.ndarray, Index, ABCSeries, ExtensionArray)):
-            if len(self) != len(other):
-                raise ValueError("Lengths must match to compare")
+        if isinstance(other, (np.ndarray, Index, ABCSeries, ExtensionArray)) and len(
```

Comment:
generally i like parens around this to group the conditions (can be a follow on )